Formula/pipenv.rb

```diff
@@ -0,0 +1,129 @@
+  end
+
+  resource "pycodestyle" do
+    url "https://files.pythonhosted.org/packages/e1/88/0e2cbf412bd849ea6f1af1f97882add46a374f4ba1d2aea39353609150ad/pycodestyle-2.3.1.tar.gz"
+    sha256 "682256a5b318149ca0d2a9185d365d8864a768a28db66a84a2ea946bcc426766"
+  end
+
+  resource "pyflakes" do
+    url "https://files.pythonhosted.org/packages/5b/b7/dcd6ebc826065ca4ccd2406aac4378e1df6eb91124625d45d520219932a1/pyflakes-1.5.0.tar.gz"
+    sha256 "aa0d4dff45c0cc2214ba158d29280f8fa1129f3e87858ef825930845146337f4"
+  end
+
+  resource "requests" do
+    url "https://files.pythonhosted.org/packages/b0/e1/eab4fc3752e3d240468a8c0b284607899d2fbfb236a56b7377a329aa8d09/requests-2.18.4.tar.gz"
+    sha256 "9c443e7324ba5b85070c4a818ade28bfabedf16ea10206da1132edaa6dda237e"
+  end
+
+  resource "shutilwhich" do
+    url "https://files.pythonhosted.org/packages/66/be/783f181594bb8bcfde174d6cd1e41956b986d0d8d337d535eb2555b92f8d/shutilwhich-1.1.0.tar.gz"
+    sha256 "db1f39c6461e42f630fa617bb8c79090f7711c9ca493e615e43d0610ecb64dc6"
+  end
+
+  resource "urllib3" do
+    url "https://files.pythonhosted.org/packages/ee/11/7c59620aceedcc1ef65e156cc5ce5a24ef87be4107c2b74458464e437a5d/urllib3-1.22.tar.gz"
+    sha256 "cc44da8e1145637334317feebd728bd869a35285b93cbb4cca2577da7e62db4f"
+  end
+
+  resource "virtualenv" do
+    url "https://files.pythonhosted.org/packages/d4/0c/9840c08189e030873387a73b90ada981885010dd9aea134d6de30cd24cb8/virtualenv-15.1.0.tar.gz"
+    sha256 "02f8102c2436bb03b3ee6dede1919d1dac8a427541652e5ec95171ec8adbc93a"
+  end
+
+  resource "virtualenv-clone" do
+    url "https://files.pythonhosted.org/packages/36/66/d95f83255138ded1aec242de5ee51536226fa7b2582a7ce0863fea5dc4f2/virtualenv-clone-0.2.6.tar.gz"
+    sha256 "6b3be5cab59e455f08c9eda573d23006b7d6fb41fae974ddaa2b275c93cc4405"
+  end
+
+  def install
+    # Using the virtualenv DSL here because the alternative of using
+    # write_env_script to set a PYTHONPATH breaks things.
+    # https://github.com/Homebrew/homebrew-core/pull/19060#issuecomment-338397417
+    venv = virtualenv_create(libexec)
+    venv.pip_install resources
+    venv.pip_install buildpath
+
+    # `pipenv` needs to be able to find `virtualenv` and `pew` on $PATH. Rather
+    # than installing symlinks for those scripts into `#{bin}` we leave them in
+    # `#{libexec}/bin` and create a wrapper script for `pipenv` which adds
+    # `#{libexec}/bin` to $PATH.
+    env = {
+      :PATH => "#{libexec}/bin:$PATH",
+    }
+    (bin/"pipenv").write_env_script(libexec/"bin/pipenv", env)
+
+    # `#{libexec}/bin` - which we've just added to $PATH - contains symlinks
+    # linking python2 and python2.7 to ./python.  We remove those symlinks so
+    # that pipenv options that search for a python version, like `pipenv --two`
+    # and `pipenv --python=2.7`, respect the user's unmodified $PATH rather than
+    # finding the copy of system python that we have just prepended onto it.
```

Comment:
s/onto/to/

Reply:
and I suppose it's not the system python itself that we're prepending ...

Reply:
> and I suppose it's not the system python itself that we're prepending

It's a virtualenv copy of the system python

Reply:
I know. I just mean that it's the directory not the file that is prepended :)